runtime/src/serde_snapshot/tests.rs

```diff
@@ -361,13 +364,192 @@ fn test_bank_serialize_newer() {
     }
 }
 
+#[test]
+fn test_extra_fields_eof() {
+    solana_logger::setup();
+    let (mut genesis_config, _) = create_genesis_config(500);
+    activate_feature(&mut genesis_config, disable_fee_calculator::id());
+
+    let bank0 = Arc::new(Bank::new_for_tests(&genesis_config));
+    bank0.squash();
+    let mut bank = Bank::new_from_parent(&bank0, &Pubkey::default(), 1);
```

Comment:
Here's where you could call `bank.activate_feature()`.

Reply:
Tried it, when it comes time to serialize the bank we call `compute_active_feature_set` and this feature set is overwritten. Seems the snapshot needs lamps in an actual account to remember the feature.